week02/Longest_Common_Prefix.cpp

```diff
@@ -0,0 +1,26 @@
+//Problem 3: Longest Common Prefix (Easy)
+
+//Solution:
+
+// Time complexity: O(n * m)
+// where, n = number of strings in the vector and
+// m = number of common characters ( length of common prefix )
+// Space complexity: O(1)
+
+string longestCommonPrefix(vector<string>& strs){
+    if(strs.size()) == 1){
+        return strs[0];
+    }
+    else{
+        int numberOfStrings = strs.size();
+        string currentLCP = strs[0];
+        for(int i = 1; i < numberOfStrings; i++){
+            int j;
+            for(int j = 0; j < strs[i].length(); j++){
+                if(strs[i][j] != currentLCP[j]) break;
+            }
+            currentLCP = currentLCP.substr(0, j);
+        }
+        return currentLCP;
+    }
+}
```

Comment:
Not the optimal one.

Reply:
Again, is the first `if` condition necessary? Don't handle special cases too often in every program you write. It's not a good practice.